iceoryx_examples/waitset/ice_waitset_basic.cpp

```diff
@@ -0,0 +1,77 @@
+#include "iceoryx_posh/popo/subscriber.hpp"
+#include "iceoryx_posh/runtime/posh_runtime.hpp"
+#include "iceoryx_posh/popo/wait_set.hpp"
+#include "iceoryx_posh/popo/user_trigger.hpp"
+#include "iceoryx_utils/posix_wrapper/signal_handler.hpp"
+#include "topic_data.hpp"
+
+#include <iostream>
+#include <atomic>
+
+std::atomic_bool shutdown{false};
+iox::popo::UserTrigger shutdownTrigger;
+
+static void sigHandler(int sig [[gnu::unused]])
+{
+    shutdown = true;
+    shutdownTrigger.trigger();
```

Comment:
I would love to avoid having to register this trigger to unblock the waitset but currently I cannot. I propose an API addition but that was not easy to do right away, given the internal logic which has certain (implicit) assumptions. Mainly expecting wait always to return a non-empty vector and ensuring this on multiple levels in the implementation which makes it error-prone/inefficient  to simply change. I propose to gather the shortcomings somewhere and rework the waitset when it is more convenient.